src/main.cpp

```diff
@@ -4478,21 +4494,27 @@ void wait_at_frontend(void)
     {
       WARNMSG("No valid mappack files found");
     }
-    //Set level number and campaign (for single level mode: GOF_SingleLevel)
-    if ((start_params.operation_flags & GOF_SingleLevel) != 0) {
-        TbBool result = false;
-        if (start_params.selected_campaign[0] != '\0') {
-            result = change_campaign(strcat(start_params.selected_campaign,".cfg"));
+    if (restart_level)
+    {
+        if (!change_campaign(cached_campaign_name)) // Load the cached campaign
+        {
+            WARNMSG("Unable to load cached campaign, can't restart the level.");
+            set_selected_level_number(first_singleplayer_level());
         }
-        if (!result) {
+        else
+        {
+            set_selected_level_number(cached_level_number); // Load the cached level
+            set_flag_byte(&game.operation_flags,GOF_SingleLevel,true); //This will load the game as if a level was specified with -level
+        }
+    }
+    //Set level number and campaign (for single level mode: GOF_SingleLevel)
+    else if ((start_params.operation_flags & GOF_SingleLevel) != 0) {
+        if (!change_campaign(strcat(start_params.selected_campaign,".cfg"))) {
             if (!change_campaign("")) {
-                WARNMSG("Unable to load default campaign for the specified level CMD Line parameter");
-            }
-            else if (start_params.selected_campaign[0] != '\0') { // only show this log message if the user actually specified a campaign
-                WARNMSG("Unable to load campaign associated with the specified level CMD Line parameter, default loaded.");
+                ERRORLOG("Unable to load default campaign for the specified level CMD Line parameter");
             }
             else {
-                JUSTLOG("No campaign specified. Default campaign loaded for selected level (%d).", start_params.selected_level_number);
+                ERRORLOG("Unable to load campaign associated with the specified level CMD Line parameter, default loaded.");
```

Comment:
Is this really an error? Won't this happen if you simply use -level without specifying the campaign?

Reply:
I'm not sure. I think I had some help doing the restart level feature.